src/glibc/nptl/pthread_create.c

```diff
@@ -340,16 +300,51 @@ static int create_thread (struct pthread *pd, const struct pthread_attr *attr,
 
   TLS_DEFINE_INIT_TP (tp, pd);
 
-  unsigned char *stack = 0;
-
-  struct clone_args *args = (void *)pd->stackblock + pd->stackblock_size - sizeof(struct clone_args) - TLS_TCB_SIZE;
+  /*
```

Comment:
I like the diagram here.

This section of code probably needs a paragraph comment explaining whats going on as well since its pretty dense.